junifer/data/masks.py

```diff
@@ -219,46 +256,74 @@ def get_mask(
             mask_name = t_mask
             mask_params = None
 
+        # If mask is being inherited from previous steps like preprocessing
         if mask_name == "inherit":
+            # Requires extra input to be passed
             if extra_input is None:
                 raise_error(
                     "Cannot inherit mask from another data item "
                     "because no extra data was passed."
                 )
+            # Missing inherited mask item
             if inherited_mask_item is None:
                 raise_error(
                     "Cannot inherit mask from another data item "
                     "because no mask item was specified "
                     "(missing `mask_item` key in the data object)."
                 )
+            # Missing inherited mask item in extra input
             if inherited_mask_item not in extra_input:
                 raise_error(
                     "Cannot inherit mask from another data item "
                     f"because the item ({inherited_mask_item}) does not exist."
                 )
             mask_img = extra_input[inherited_mask_item]["data"]
+        # Starting with new mask
         else:
-            mask_object, _ = load_mask(
+            # Load mask
+            mask_object, _, mask_space = load_mask(
                 mask_name, path_only=False, resolution=resolution
             )
+            # If mask is callable like from nilearn
             if callable(mask_object):
                 if mask_params is None:
                     mask_params = {}
                 mask_img = mask_object(target_img, **mask_params)
-            else:  # Mask is a Nifti1Image
+            # Mask is a Nifti1Image
+            else:
+                # Mask params provided
                 if mask_params is not None:
+                    # Unused params
                     raise_error(
                         "Cannot pass callable params to a non-callable mask."
                     )
+                # Resample mask to target image
                 mask_img = resample_to_img(
                     mask_object,
                     target_img,
                     interpolation="nearest",
                     copy=True,
                 )
+            all_spaces.append(mask_space)
         all_masks.append(mask_img)
+
+    # Multiple masks, need intersection / union
     if len(all_masks) > 1:
-        mask_img = intersect_masks(all_masks, **intersect_params)
+        # Intersect / union of masks only if all masks are in the same space
+        if len(set(all_spaces)) == 1:
+            mask_img = intersect_masks(all_masks, **intersect_params)
+        else:
+            # Check for inherited masks with target image in MNI space
+            if "inherit" in all_spaces and target_data["space"].startswith(
```

Comment:
I think at this point we should not have the "inherit" as it was already computed before. Usually the "inherit" is a callable.